### PR TITLE
Requests console helpers

### PR DESCRIPTION
**Pull request by Profakos**
## About The Pull Request

This PR adds mapping helpers for requests consoles. Two of them set up if the console can receive ore updates, and if it can make announcements, flipping their relevant variables to TRUE. The other three adds the consoles to their relevant department console lists during late initialize. This allowed me to remove three variables directly from the consoles themselves.

image
New sprites by CoiledLamb!

~~This PR also anchors mapping helpers, to prevent effects like the roundstart crate initialization from moving them.~~ This was fixed by a different PR.

## Why It's Good For The Game

Less var edits, easier to see a console's type at a glance.

## Changelog

:cl: Profakos, sprites by CoiledLamb
qol: Most request console varedits have been moved to mapping helpers.
/:cl:

### CODE REVIEW

**Comment by Profakos** [comment]
> Runtime in mapping_helpers.dm,1241: Cannot read null.type
  proc name: LateInitialize (/obj/effect/mapping_helpers/requests_console/LateInitialize)
  src: the request console supplies r... (/obj/effect/mapping_helpers/requests_console/supplies)
  src.loc: the science crate (/obj/structure/closet/crate/science)
  call stack:
  the request console supplies r... (/obj/effect/mapping_helpers/requests_console/supplies): LateInitialize(null)

A crate ate my mapping helper??

**Comment by Profakos** [comment]
image

Yeah its the anchored crate in deltastation robotics. Apparently crates suck up everything under atom/movable, so it will eat mapping helpers too. I am afraid moving the console would just mask the issue.

**Comment by Profakos** [comment]
I elected to do the simplest solution for now, which is anchoring mapping helpers, same way as jobstart landmarks are anchored. I don't want to increase the scope of this PR to refactor how crates eating items during initialization works.
In the meanwhile, @CoiledLamb offered to make new sprites for the mapping helpers, so I am leaving it drafted until they are ready.

**Comment by Profakos** [comment]
New sprites are done! Ready for review.

File: code/modules/mapping/mapping_helpers.dm [hunk 1]
```diff
@@ -1220,3 +1220,71 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
...
+/obj/effect/mapping_helpers/requests_console/proc/payload(obj/machinery/requests_console/console)
+	return
+
+/obj/effect/mapping_helpers/requests_console/announcement
+	name = "request console announcement helper"
+	icon_state = "requests_console_announcement_helper"
+
+/obj/effect/mapping_helpers/requests_console/announcement/payload(obj/machinery/requests_console/console)
+	console.can_send_announcements = TRUE
+	return
```
Comment (san7890) [inline code/modules/mapping/mapping_helpers.dm:1258]:
```suggestion
```

redundant

File: code/modules/mapping/mapping_helpers.dm [hunk 2]
```diff
@@ -1220,3 +1220,71 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
...
+
+/obj/effect/mapping_helpers/requests_console/assistance
+	name = "request console assistance requestable helper"
+	icon_state = "requests_console_assistance_helper"
+
+/obj/effect/mapping_helpers/requests_console/assistance/payload(obj/machinery/requests_console/console)
+	GLOB.req_console_assistance |= console.department
+	return
```
Comment (san7890) [inline code/modules/mapping/mapping_helpers.dm:1266]:
```suggestion
```

File: code/modules/mapping/mapping_helpers.dm [hunk 3]
```diff
@@ -1220,3 +1220,71 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
...
+
+/obj/effect/mapping_helpers/requests_console/supplies
+	name = "request console supplies requestable helper"
+	icon_state = "requests_console_supplies_helper"
+
+/obj/effect/mapping_helpers/requests_console/supplies/payload(obj/machinery/requests_console/console)
+	GLOB.req_console_supplies |= console.department
+	return
```
Comment (san7890) [inline code/modules/mapping/mapping_helpers.dm:1274]:
```suggestion
```

File: code/modules/mapping/mapping_helpers.dm [hunk 4]
```diff
@@ -1220,3 +1220,71 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
...
+
+/obj/effect/mapping_helpers/requests_console/information
+	name = "request console information relayable helper"
+	icon_state = "requests_console_information_helper"
+
+/obj/effect/mapping_helpers/requests_console/information/payload(obj/machinery/requests_console/console)
+	GLOB.req_console_information |= console.department
+	return
```
Comment (san7890) [inline code/modules/mapping/mapping_helpers.dm:1282]:
```suggestion
```

File: code/modules/mapping/mapping_helpers.dm [hunk 5]
```diff
@@ -1220,3 +1220,71 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
...
+
+/obj/effect/mapping_helpers/requests_console/ore_update
+	name = "request console ore update helper"
+	icon_state = "requests_console_ore_update_helper"
+
+/obj/effect/mapping_helpers/requests_console/ore_update/payload(obj/machinery/requests_console/console)
+	console.receive_ore_updates = TRUE
+	return
```
Comment (san7890) [inline code/modules/mapping/mapping_helpers.dm:1290]:
```suggestion
```

**Review by san7890 — CHANGES_REQUESTED** [review]
I was wondering when someone would finally take care of these! I'll just assume direct control since these are just very nitty

**Pull request merged**